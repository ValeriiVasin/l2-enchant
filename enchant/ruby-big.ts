import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const bigRubyEnchantMap = new Map([
  ...createEnchantMap({
    strategy: EnchantStrategy.Drop,
    enchantItem: Item.RubyEnergy,
    items: [
      Item.RubyBig_1,
      Item.RubyBig_2,
      Item.RubyBig_3,
      Item.RubyBig_4,
      Item.RubyBig_5,
    ],
    chances: [35, 25, 15, 10],
  }),
]);
