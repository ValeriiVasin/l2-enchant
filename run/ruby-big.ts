import { rubyBigEnchantMap } from '../enchant/ruby-big';
import { itemsRange } from '../helpers/items-range';
import { run } from '../helpers/run';
import { Item } from '../items';

run(Item.RubyBig_1, itemsRange(Item.RubyBig_2, Item.RubyBig_5), 100_000, rubyBigEnchantMap);
