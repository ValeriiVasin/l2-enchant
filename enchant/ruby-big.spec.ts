import { rubyBigEnchantMap } from './ruby-big';

it('enchant map: ruby big', () => {
  expect(rubyBigEnchantMap).toMatchSnapshot();
});
