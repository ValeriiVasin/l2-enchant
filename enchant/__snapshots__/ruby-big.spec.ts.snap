// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: ruby big 1`] = `
Map {
  "Большой Рубин Ур.1" => Object {
    "cost": "Энергия Самоцвета",
    "item": "Большой Рубин Ур.1",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Большой Рубин Ур.2",
      },
      Object {
        "chance": 65,
        "item": "Большой Рубин Ур.1",
      },
    ],
  },
  "Большой Рубин Ур.2" => Object {
    "cost": "Энергия Самоцвета",
    "item": "Большой Рубин Ур.2",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Большой Рубин Ур.3",
      },
      Object {
        "chance": 75,
        "item": "Большой Рубин Ур.1",
      },
    ],
  },
  "Большой Рубин Ур.3" => Object {
    "cost": "Энергия Самоцвета",
    "item": "Большой Рубин Ур.3",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Большой Рубин Ур.4",
      },
      Object {
        "chance": 85,
        "item": "Большой Рубин Ур.1",
      },
    ],
  },
  "Большой Рубин Ур.4" => Object {
    "cost": "Энергия Самоцвета",
    "item": "Большой Рубин Ур.4",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Большой Рубин Ур.5",
      },
      Object {
        "chance": 90,
        "item": "Большой Рубин Ур.1",
      },
    ],
  },
}
`;
